### PR TITLE
Add Helm option for using separate Commands and Args in sidecar container

### LINKED ISSUE #5023: Support Dapr deployments on ACI virtual kubelets
## In what area(s)?

/area runtime

> /area operator

> /area placement

> /area docs

> /area test-and-release

## Describe the feature

**Background**
With https://github.com/dapr/dapr/pull/4741 we modified a behavior for our k8s deployment. 
- Originally, both Command and Args were used, where Command would be `daprd` or `dlv` and Args would be the appropriate arguments. 
- After the above PR, Command is no longer used. The Args now contain `daprd` or `dlv` followed by other arguments. This allows using custom ENTRYPOINT in Docker images. For example, the Windows image now uses a custom ENTRYPOINT `setup-certificates.cmd` that installs certificates in the Windows container if necessary. For more, see [docs](https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/#running-a-command-in-a-shell).

**Problem**
This broke the ACI virtual-kubelet deployments. Root cause of the issue can be found here https://github.com/virtual-kubelet/azure-aci/issues/10. It is because of the way ACI API is structured, it expects Command to be always set. 

Error message:
> Type Reason Age From Message
Normal Scheduled 10s default-scheduler Successfully assigned backend-test/nginx-deployment-67f6b7c46-qvbnp to virtual-node-aci-linux
Warning ProviderCreateFailed 4s (x10 over 10s) virtual-node-aci-linux/pod-controller ACI does not support providing args without specifying the command. Please supply both command and args to the pod spec.

For more details on the error, see the comments in #4741.

**Solution**
We can add a special handling for ACI virtual-kubelet, and have Dapr specify both Command and Args. Note, this will cause ENTRYPOINTs in Docker images to no longer work and will be a limitation. The immediate side-effect will be that [certificate installation](https://docs.dapr.io/operations/configuration/install-certificates/) won't be allowed on Windows nodes in ACI virtual-kubelet.

## Release Note

RELEASE NOTE: **ADD** Support Dapr deployment in ACI virtual-kubelet.

### PR DESCRIPTION
**Pull request by shubham1172**
Signed-off-by: Shubham Sharma <shubhash@microsoft.com>

# Description

This PR introduces a new Helm chart configuration in the dapr sidecar injector - `ignoreEntrypointTolerations`. It is a JSON array of tolerations.

If the application pod contains any of the tolerations specified by the Helm config above, the sidecar image has an explicit Command set (`daprd` or `dlv`) , along with the correct Args. Otherwise, Command is kept as empty and `daprd` or `dlv` is a part of Args. This allows the ENTRYPOINTS in Dapr Docker image to be executed first.

The motivation for this PR is to support cloud providers (e.g. ACI virtual-kubelet) that mandate both Command and Args to be set. In these cases, the ENTRYPOINT of Docker images are overridden. See the attached issue for more details on this.   

## Note to reviewers

Please refer to existing comments in #5024

## Issue reference

Please reference the issue this PR will close #5023

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [X] Created/updated tests
* [X] Unit tests passing
* [X] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: charts/dapr/README.md [hunk 1]
```diff
@@ -169,6 +169,7 @@ The Helm chart has the follow configuration options that can be supplied:
 | `dapr_sidecar_injector.resources`         | Value of `resources` attribute. Can be used to set memory/cpu resources/limits. See the section "Resource configuration" above. Defaults to empty | `{}` |
 | `dapr_sidecar_injector.debug.enabled`     | Boolean value for enabling debug mode | `{}` |
 | `dapr_sidecar_injector.kubeClusterDomain` | Domain for this kubernetes cluster. If not set, will auto-detect the cluster domain through the `/etc/resolv.conf` file `search domains` content. | `cluster.local` |
+| `dapr_sidecar_injector.ignoreEntrypointTolerations` | JSON array of Kubernetes tolerations. If pod contains all the tolerations, it will ignore the Docker image ENTRYPOINT for Dapr sidecar. | `[{\"effect\":\"NoSchedule\",\"key\":\"azure.com/aci\"}]` |
```
Comment (artursouza) [inline charts/dapr/README.md:172]: Please, we need only one to match and not all.

**Review by artursouza — COMMENTED** [review]
Initial feedback. Need more time.

File: charts/dapr/README.md (hunk 1 above)
Comment (shubham1172) [reply charts/dapr/README.md]: Thanks Artur, this has been updated!

**Comment by shubham1172** [comment]
# Test results

To reproduce, setup ACI VK on AKS using [Create and configure an Azure Kubernetes Services (AKS) cluster to use virtual nodes in the Azure portal](https://docs.microsoft.com/en-in/azure/aks/virtual-nodes-portal), and deploy [hello-kubernetes](https://github.com/dapr/quickstarts/blob/master/tutorials/hello-kubernetes/deploy/python.yaml) with the correct nodeSelector and tolerations from [Schedule a pod in ACI](https://github.com/virtual-kubelet/azure-aci#schedule-a-pod-in-aci).

## Using Dapr 1.8.4

```
Events:
  Type     Reason                Age                 From                                   Message
  ----     ------                ----                ----                                   -------
  Normal   Scheduled             38s                 default-scheduler                      Successfully assigned default/pythonapp-5f7df7b8ff-d9cld to virtual-node-aci-linux
  Warning  ProviderCreateFailed  16s (x12 over 37s)  virtual-node-aci-linux/pod-controller  ACI does not support providing args without specifying the command. Please supply both command and args to the pod spec.
```

## Using Dapr from this branch

```bash
dapr status -k                                                                                                  

  NAME                   NAMESPACE    HEALTHY  STATUS   REPLICAS  VERSION              AGE  CREATED
  dapr-sentry            dapr-system  True     Running  1         aci-fix-linux-amd64  23s  2022-08-24 16:17.58
  dapr-placement-server  dapr-system  True     Running  1         aci-fix-linux-amd64  24s  2022-08-24 16:17.58
  dapr-operator          dapr-system  True     Running  1         aci-fix-linux-amd64  24s  2022-08-24 16:17.58
  dapr-sidecar-injector  dapr-system  True     Running  1         aci-fix-linux-amd64  24s  2022-08-24 16:17.58
  dapr-dashboard         dapr-system  True     Running  1         0.10.0               25s  2022-08-24 16:17.57
```

```
k describe pod pythonapp-5f7df7b8ff-mmvql

Events:
  Type    Reason                 Age   From                                   Message
  ----    ------                 ----  ----                                   -------
  Normal  Scheduled              2m7s  default-scheduler                      Successfully assigned default/pythonapp-5f7df7b8ff-mmvql to virtual-node-aci-linux
  Normal  ProviderCreateSuccess  2m4s  virtual-node-aci-linux/pod-controller  Create pod in provider successfully
```

```
k logs pythonapp-5f7df7b8ff-mmvql -c daprd

time="2022-08-24T10:53:00.312183336Z" level=info msg="starting Dapr Runtime -- version edge -- commit 9b07bc23e7321b609b44fe31c554d25a2a4cdcd0" app_id=pythonapp instance=wk-caas-206d04be62d04735a136a044903411f0-96ebe8db1b5f9b5e896f4e scope=dapr.runtime type=log ver=edge
```

**Comment by shubham1172** [comment]
PR is ready for review. 

/cc @ItalyPaleAle @artursouza @yaron2

**Review by ItalyPaleAle — DISMISSED** [review]

**Review by yaron2 — DISMISSED** [review]
Good PR! A big improvement to the hard-coded iteration.

**Review by yaron2 — APPROVED** [review]

**Pull request merged**